src/worker/plugins/run.ts

```diff
@@ -1,9 +1,15 @@
 import { PluginEvent } from '@posthog/plugin-scaffold'
 
 import { processError } from '../../shared/error'
+import { posthog } from '../../shared/posthog'
 import { PluginConfig, PluginsServer } from '../../types'
 
+const EVENTS_TO_IGNORE = ['$plugin_running_duration']
+
 export async function runPlugins(server: PluginsServer, event: PluginEvent): Promise<PluginEvent | null> {
+    if (EVENTS_TO_IGNORE.includes(event.event)) {
+        return event
+    }
```

Comment:
Hot tip, if you ever find yourself building escape hatches around large blocks of code, it's probably a [code smell](https://en.wikipedia.org/wiki/Code_smell) and something's off with the abstraction or implementation.

In this case, we send the `$plugin_running_duration` event to app.posthog.com, not to the self-hosted instance that's processing events. Hence we don't really have much do escape here.

Reply:
Hmm but app will also be sending `$plugin_running_duration` to app, no?

Reply:
I guess you had the escape hatch here because every event generated another event and hence without it (and with the posthog.capture below) we would get stuck in an endless loop.

If we send this event only hourly or otherwise periodically, it won't matter if we also measure how long a few plugins run on it.

Reply:
Yeah exactly, my comment was related to your above comment but with the new approach this is irrelevant